ddev/src/ddev/cli/meta/scripts/upgrade_python.py

```diff
@@ -20,57 +21,200 @@ def upgrade_python(app: Application, version: str):
     \b
     `$ ddev meta scripts upgrade-python 3.11`
     """
-    import tomlkit
 
     from ddev.repo.constants import PYTHON_VERSION as old_version
 
     tracker = app.create_validation_tracker('Python upgrades')
 
     for target in app.repo.integrations.iter_testable(['all']):
-        config_file = target.path / 'hatch.toml'
-        test_config = tomlkit.parse(config_file.read_text())
+        update_hatch_file(app, target.path, version, old_version, tracker)
+        update_pyproject_file(target, version, old_version, tracker)
```

Comment:
Given the number of functions with an identical signature, you could consider having a class that accepts all this data at initialization. Would reduce the amount of boilerplate at the cost of having an unclear abstraction (`TaskContext`?).

Reply:
That's true, but that would mean two different classes since we have a part that upgrades integrations and a part that upgrades global stuff (such as ci). So that would mean at least two unclear abstractions. Given that's a script that's not going to be used a lot (I don't even expect to use it once a year) I think it's ok to keep it as it is. What do you think?